sympy/sets/handlers/functions.py

```diff
@@ -66,6 +66,10 @@ def _set_function(f, x): # noqa:F811
     if not x.start.is_comparable or not x.end.is_comparable:
         return
 
+    # XXX: when non-rational functions are supported, remove this line. 
```

Comment:
```suggestion
    # XXX: when non-rational functions are supported, remove this line.
```

Reply:
If this is needed then the trailing space must be removed. (There is a setting for git that can be used to do this automatically, btw.)

But why is this needed?

Reply:
When `singularities` was not updated, this function would just return None for non-rational functions, as `singularities` would throw a `NotImplementedError`. I've just replicated the same behaviour as making this function support non-rational functions may take some non-trivial effort and I think this is best done in another PR (for a start, `singularities` needs to be replaced by `continuous_domain` or even some function like `function_domain`)

Reply:
I pushed a commit to my `cd` branch that relaxes this constraint and fixes issue #12560